File: cmstestsuite/tasks/communication2/code/stub.pas
program stub;
uses sysutils, user1, user2;

var
    n: longint;
    infile, outfile: text;
    procid: longint;
    buf: array[1..1] of byte;

begin
    assign(infile, ParamStr(1));
    reset(infile);
    assign(outfile, ParamStr(2));
    rewrite(outfile);
    (* The output must be unbuffered - this seems to work. *)
    settextbuf(outfile, buf, 1);

    procid := StrToInt(ParamStr(3));

    while True do
    begin
        readln(infile, n);
        if (n = 0) then
            break;
        write(outfile, 'correct ');
        if (procid = 0) then
            writeln(outfile, userfunc1(n))
        else
            writeln(outfile, userfunc2(n));
        flush(outfile);
    end;
    close(infile);
    close(outfile);
end.

File: cmstestsuite/tasks/communication_many_fifoio_stubbed/code/stub.py
import sys

import user1
import user2


fin = open(sys.argv[1], "r")
fout = open(sys.argv[2], "w")

while True:
    n = fin.readline().strip()
    if n == "":
        # EOF, let's quit.
        break

    n = int(n)
    if n == 0:
        break
    if int(sys.argv[3]) == 0:
        fout.write("correct %d\n" % user1.userfunc1(n))
    else:
        fout.write("correct %d\n" % user2.userfunc2(n))
    fout.flush()
